examples/large_files/test/tests.ts

```diff
@@ -0,0 +1,126 @@
+            test: async () => {
+                await new Promise((resolve) => setTimeout(resolve, 10 * 1000));
+                return { Ok: true };
+            }
+        },
+        // Permanent Assets
+        generateTest(
+            'photos/people/george-washington.tif',
+            'permanent',
+            origin
+        ),
+        generateTest('photos/places/dinosaurNM.jpg', 'permanent', origin),
+        generateTest('photos/places/slc.jpg', 'permanent', origin),
+        generateTest('photos/things/book.jpg', 'permanent', origin),
+        generateTest('photos/things/utah-teapot.jpg', 'permanent', origin),
+        generateTest(
+            'text/subfolder/deep-sub-folder/deep.txt',
+            'permanent',
+            origin
+        ),
+        generateTest(
+            'text/subfolder/sibling-deep-sub-folder/deep.txt',
+            'permanent',
+            origin
+        ),
+        generateTest('text/subfolder/other-thing.txt', 'permanent', origin),
+        generateTest('text/thing.txt', 'permanent', origin),
+        generateTest('text/thing.txt', 'permanent', origin),
+        generateTest('text/single.txt', '', origin, 'single_asset.txt'),
+
+        // Auto Generated Assets
+        //      Edge Cases
+        { ...generateTest('test0B', 'auto', origin), skip: true }, // TODO we have problems with 0B files on the canister side
+        generateTest('test1B', 'auto', origin),
+        generateTest(`test${120 * 1024 * 1024 + 1}B`, 'auto', origin),
+        generateTest(`test${150 * 1024 * 1024 + 1}B`, 'auto', origin),
+        generateTest('test2000001B', 'auto', origin),
+        //      General Cases
+        generateTest('test1KiB', 'auto', origin),
+        generateTest('test10KiB', 'auto', origin),
+        generateTest('test100KiB', 'auto', origin),
+        generateTest('test1MiB', 'auto', origin),
+        generateTest('test10MiB', 'auto', origin),
+        generateTest('test100MiB', 'auto', origin),
+        generateTest('test250MiB', 'auto', origin),
+        { ...generateTest('test500MiB', 'auto', origin), skip: true }, // We currently run out of memory with this file
+        { ...generateTest('test1GiB', 'auto', origin), skip: true }
+        // generateTest(`test${2_000_000 * 18}B`, 'auto', origin),
+        // generateTest(`test${2_000_000 * 18 + 1}B`, 'auto', origin)
+    ];
+}
+
+function generateTest(
+    fileName: string,
+    nodeDir: string,
+    origin: string,
+    nodeName?: string
+): Test {
+    return {
+        name: `upload: ${fileName}`,
+        test: async () => {
+            const canisterFilePath = `assets/${fileName}`;
+            const dir = nodeDir === '' ? '' : `/${nodeDir}`;
+            const nodeFileName = nodeName ?? fileName;
+            const nodeFilePath = `assets${dir}/${nodeFileName}`;
+            const actor = await createGetFileHashActor(
+                getCanisterId('large_files'),
+                '8000'
```

Comment:
get the port number through automatic means...maybe just make a get agent host function